src/pydvl/value/results.py

```diff
@@ -0,0 +1,232 @@
+"""
+This module collects types and methods for the inspection of the results of
+valuation algorithms.
+
+The most important class is :class:`ValuationResult`, which provides access
+to raw values, as well as convenient behaviour as a Sequence with extended
+indexing abilities, and conversion to `pandas DataFrames
+<https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html>`_.
+"""
+import collections.abc
+from dataclasses import dataclass
+from enum import Enum
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Generator,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Union,
+    cast,
+    overload,
+)
+
+import numpy as np
+
+from pydvl.utils import Dataset, SortOrder
+
+try:
+    import pandas  # Try to import here for the benefit of mypy
```

Comment:
Pandas is part of the requirements. I don't see how this import could fail.

Reply:
Good catch. But I don't think it should be a requirement. As a matter of fact, I see it's only really used in Dataset, and that could all be put into a factory `Dataset.from_dataframe()`, making pandas optional. So I'll leave this  here and create #218